local.tf

```diff
@@ -103,6 +103,91 @@ locals {
     var.workers_group_defaults,
   )
 
+  volume_defaults = {
+    volume_size           = local.workers_group_defaults.root_volume_size       # root volume size of workers instances.
+    volume_type           = local.workers_group_defaults.root_volume_type       # root volume type of workers instances, can be 'standard', 'gp2', or 'io1'
+    iops                  = local.workers_group_defaults.root_iops              # The amount of provisioned IOPS. This must be set with a volume_type of "io1".
+    volume_throughput     = local.workers_group_defaults.root_volume_throughput # The amount of throughput to provision for a gp3 volume.
+    block_device_name     = local.workers_group_defaults.root_block_device_name # Root device name for workers. If non is provided, will assume default AMI was used.
+    encrypted             = local.workers_group_defaults.root_encrypted         # Whether the volume should be encrypted or not
+    kms_key_id            = local.workers_group_defaults.root_kms_key_id        # The KMS key to use when encrypting the root storage device
+    delete_on_termination = true
+  }
+
+  worker_groups_launch_template_with_defaults = {
+    for key, template in var.worker_groups_launch_template : template.name => merge(
+      local.workers_group_defaults,
+      { original_index = key },
+      merge(
+        template,
+        {
+          additional_ebs_volumes = [
+            for volume_definition in template.additional_ebs_volumes : merge(
+              local.volume_defaults,
+              volume_definition
+            )
+          ]
+        }
+      )
+    )
+  }
+
+  worker_groups_with_defaults = {
+    for key, group in var.worker_groups : group.name => merge(
+      local.workers_group_defaults,
+      { original_index = key },
+      merge(
+        group,
+        {
+          additional_ebs_volumes = [
+            for volume_definition in group.additional_ebs_volumes : merge(
+              local.volume_defaults,
+              volume_definition
+            )
+          ]
+        }
+      )
+    )
+  }
+
+  # These sorted arrays are used in outputs to keep the original input order intact
+  sorted_aws_autoscaling_group_workers = values({
+    for key, value in aws_autoscaling_group.workers :
+    format("%04d", local.worker_groups_with_defaults[key].original_index) => value
+  })
```

Comment:
Could these be e.g.:
```
[ for g in var.worker_groups : aws_autoscaling_group.worker[g.name] ]
```

Reply:
youre right. with the appropriate conditions at first this works and is a bit cleaner